### ISSUE TITLE
Dev: Update <gcds-checkbox> into <gcds-checkboxes>

### ISSUE THREAD

**Issue by daine** [cds-snc/design-gc-conception#1143]
As part of the fieldset alignment work, `gcds-checkbox` will be reworked and renamed into the `gcds-checkboxes` component. `gcds-checkboxes` will remove the dependancy of being wrapped in a `gcds-fieldset` and incorporate that functionality into the checkboxes component so it functions similarly to the other form components. `gcds-checkboxes` will be able to render a group of checkboxes within a fieldset or a single checkbox without a fieldset.

## ✅ Definition of Done / Outcomes
- [x] Done when tests have been updated to include check for new properties
- [x] Done when storybook has been updated to include new properties
- [x] Done when storybook contains examples of the new properties
- [ ] Done when PR has been approved (not merged nor released yet)
- [ ] PR to remove `gcds-checkbox` from component library has been opened.

## 📜 Acceptance criteria
- [x] A property named `legend` can be set within `<gcds-checkboxes>`
- [x] A property named `options` can be set within `<gcds-checkboxes>`
- [x] Property validation for required properties
- [x] Multiple checkboxes can be rendered with the component

## 📝 More info
We have originally been recommending to use <gcds-fieldset> to wrap radio buttons (<radio-group>) and checkboxes (checkbox). Unfortunately, this same HTML element provides multiple purposes such as grouping multiple form components, as well as being the primary label for a group of form choices in radio buttons or checkboxes.

**. 🚫 Out of scope**
Other parts of the fieldset alignment work have tickets for them and is not within the scope of this ticket.

## ⚠️ Blocker
Requires #1144  . cards being closed. blocker is unblocked !
